### ISSUE TITLE
Feature flag for dynamic linking

### ISSUE THREAD

**Issue by KAAtheWiseGit** [nushell/nushell#13730]
### Related problem

To reduce the size of Alpine Linux port of Nushell, it's dynamically linked.  This, however, requires a minor patch, which enables the `sqlite-dynlib` feature in `reedline` and disables the `bundled` feature in `rusqlite`.

### Describe the solution you'd like

It would be nice to have an optional build feature, which would enable dynamic linking of SQLite.

**Comment by sholderbach** [comment]
Reminder that rust/cargo features are additive, so a feature can not disable another feature. would have to be broken up into a positive feature for statically linking sqlite, which would be part of the default feature set (for reliable distribution as self contained binaries).

**Comment by sholderbach** [comment]
(narrowed the scope of the issue title)

**Comment by KAAtheWiseGit** [comment]
Yeah, I decided to try and make a patch, and hit a wall with the additive thing.  I thought it'd be a simple side change, but it'll require adding a new default feature set.  Unsure how acceptable it is to the team

**Comment by sholderbach** [comment]
Currently the default feature set is not yet obnoxiously complicated that we can tolerate an addition. The only downside for including `static-sqlite` in the default set, would be that the requirements for `cargo build/test --no-default-features` would necessarily increase (you as a contributor have to provide a recent sqlite on the system). I think disabling the other features left has minimal impact on the things that are actively tested (and `openssl` should be much more prevalent on linux distros)
